client/blocks/qr-code-login/index.jsx

```diff
@@ -0,0 +1,266 @@
+import { getTracksAnonymousUserId } from '@automattic/calypso-analytics';
+import config from '@automattic/calypso-config';
+import { Card } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import QRCode from 'qrcode.react';
+import { useEffect, useState, useMemo } from 'react';
+import qrCenter from 'calypso/assets/images/qr-login/wp.png';
+import ExternalLink from 'calypso/components/external-link';
+import { setStoredItem, getStoredItem } from 'calypso/lib/browser-storage';
+import { useInterval } from 'calypso/lib/interval';
+
+import './style.scss';
+
+const AUTH_PULL_INTERVAL = 5000; // 5 seconds
+const LOCALE_STORAGE_KEY = 'qr-login-token';
+
+const isStillValidToken = ( tokenData ) => {
+	if ( ! tokenData?.expires ) {
+		return false;
+	}
+	return tokenData.expires > Date.now() / 1000;
+};
+
+const getLoginActionResponse = async ( action, args ) => {
+	const url = `https://wordpress.com/wp-login.php?action=${ action }`;
+	const body = new URLSearchParams( {
+		...args,
+		client_id: config( 'wpcom_signup_id' ),
+		client_secret: config( 'wpcom_signup_key' ),
+	} );
+
+	const response = await fetch( url, {
+		method: 'POST',
+		credentials: 'include',
+		body,
+	} );
+
+	let json_response = null;
+	try {
+		json_response = await response.json();
+	} catch {
+		json_response = false;
```

Comment:
Interesting that the return value is `false` when `response.json()` fails, but when the `fetch` fails, the function returns a rejected promise. That looks like inconsistent error handling, was that really your intent?